File: Qt_turtle_service/demo/ROS_demo/msg_demo/src/student_pubs.cpp
#include "ros/ros.h"
#include "msg_demo/Student.h"

int main(int argc, char *argv[])
{
    /* code */
    setlocale(LC_ALL,"");
    ros::init(argc, argv, "people_pubs");

    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<msg_demo::Student>("people_msgs", 1000);

    msg_demo::Student p;
    p.name = "Qiao";
    p.age = 27;
    p.height = 1.71;

    ros::Rate r(1);

    while (ros::ok())
    {
        /* code */
        pub.publish(p);
        p.age += 1;

        ROS_INFO("我叫:%s,今年%d岁,高%.2f米", p.name.c_str(), p.age, p.height);

        r.sleep();

        ros::spinOnce();
    }
    

    return 0;
}

File: ROS_demo/msg_demo/scripts/student_pubs.py
#! /usr/bin/env python

import rospy
from msg_demo.msg import Student

if __name__ == "__main__":

    publisher_node = 'person_pub'
    rospy.init_node(publisher_node)

    topic_name = 'person_msgs'
    pubs = rospy.Publisher(topic_name, Student, queue_size=1000)

    rate = rospy.Rate(2)

    count = 0
    while not rospy.is_shutdown():

        person = Student()
        person.name = 'Qiao'
        person.age = 27
        person.height = 1.72

        pubs.publish(person)

        count += 1
        rate.sleep()
    
    